source/gravity_model/radial.cc

```diff
@@ -178,6 +183,159 @@ namespace aspect
                                "do not have either a spherical or ellipsoidal natural coordinate system."));
 
     }
+
+
+
+// ----------------------------- RadialLinearWithTidalPotential ----------------------
+
+
+    template <int dim>
+    Tensor<1,dim>
+    RadialLinearWithTidalPotential<dim>::gravity_vector (const Point<dim> &p) const
+    {
+      /**
+       * This gravity model is a plugin for a case where the
+       * tidal potential by flattening and non-synchronnous rotation changes gravity with position and time.
+       *
+       * The equation implemented in this heating model is from Tobie et al. (2025) (https://doi.org/10.1007/s11214-025-01136-y),
+       * which is defined as:
+       * g = -magnitude - gradient (-tidal potential)).
+       * potential = (3 G M_p) / (2 a_s^3) * r^2 * (Tstar + T0)
+       * Tstar = 1/6 *(1-3*cos(theta)^2) and T0=1/2sin(theta)^2*cos(2*lambda + 2*b*t)
+       * where G = gravitational constant, M_p = mass of planet, a_s = semimajor axis of satellite's orbit, b = angular rate of nonsynchronous rotation.
+       * r, theta and lambda are radial distance, polar angle and azimuthal angle, respectively.
+       * b [1/s] = b_NSR [m/yr] / (circumference of satellite [m]) / year_to_seconds [s/yr] * 2 * pi
```

Comment:
This part of the explanation needs to go to the header file where I ask for documentation about this class. It is a high level summary of what this plugin is for.

The part below my comment (about conversion to Cartesian coordinates) can stay here, it is only relevent for the implementation, but not for understanding what the purpose of this plugin is.